pkg/cri/server/sandbox_run.go

```diff
@@ -95,6 +84,25 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		},
 	)
 
+	// cleanupErr records the last error created by the critical cleanup in defer functions,
+	// like CNI teardown and stop running sandbox task.
+	// So when the cleanup is not completed for some reason, the CRI-plugin will remain the sandbox
+	// in the not-ready state, which can be cleanup by kubelet syncPod.
```

Comment:
Kubelet `syncPod` doesn't clean low-level resources directly.  It handles that by calling `StopPodSandbox`.

In Kubelet, `syncPod` lists current sandboxes, notices the target sandbox exists, removes the sandbox and re-create it. The event message is `Pod sandbox changed, it will be killed and re-created`.